src/Chunk.ts

```diff
@@ -1021,13 +1034,22 @@ export default class Chunk {
 
 	private getChunkExportDeclarations(): ChunkExports {
 		const exports: ChunkExports = [];
+		const exportsNames = new Set<string>();
 		for (const exportName of this.getExportNames()) {
 			if (exportName[0] === '*') continue;
 
-			const variable = this.exportNames[exportName];
+			let variable = this.exportNames[exportName];
 			const module = variable.module;
-
 			if (module && module.chunk !== this) continue;
+
+			// Exports of synthetic named exports only become real if
+			// the chunk is a entry-point, otherwise we render the "default" export
+			const isEntryPoint = !!(this.facadeModule && this.facadeModule.isUserDefinedEntryPoint);
```

Comment:
I still think that `isUserDefinedEntryPoint` should not matter at all to the logic. A chunk is an entry chunk and uses entry chunk semantics if and only if `this.facadeModule` is set. `this.isUserDefinedEntryPoint` is only important to decide which naming scheme the chunk should follow. I did not manage to write a broken test for this that did not trigger other issues yet, though.